src/audio/module_adapter/module/volume/volume.c

```diff
@@ -293,49 +291,39 @@ static void volume_ramp(struct processing_module *mod)
 
 	/* Current ramp time in Q29.3 milliseconds. Note that max. ramp length
 	 * can be 1.3s at 192 kHz rate and 5.5s at 48 kHz rate without
-	 * exceeding int32_t range. Value 8000 is 1000 for converting to
-	 * milliseconds times 8 (2 ^ 3) for fraction.
+	 * exceeding int32_t range. Inverse of sample rate is 1000/sample_rate
+	 * for milliseconds.
```

Comment:
I am not quite understanding why ramp time still need a  Q format, possible to set ramp time with a fixed like 20-50ms? ramp functionality is for reduce pop noise and make volume change smooth, if we have fixed time, then based on sample rate, we will know how many samples need process with ramp, then a counter will be set, it would be much simpler.

Reply:
Ramp is relative time and independent from stream rate. If the linear and windows ramp would take sample count as parameter they would any way need to compensate the ramp step to stream sample rate. Also the ramp update rate is independent of sample rate. The fastest ramp update is 8 kHz (to avoid zipper noise) independently of stream rate. But we have also 2 kHz and 1 kHz ramp updates.

Reply:
So, in other words, simplification of this is hard.